crates/core/tedge_api/src/messages.rs

```diff
@@ -473,65 +477,115 @@ impl From<SoftwareError> for Option<SoftwareModuleItem> {
     }
 }
 
-#[derive(Debug, Deserialize, Serialize, Eq, PartialEq)]
-pub enum RestartOperation {
-    Request(RestartOperationRequest),
-    Response(RestartOperationResponse),
-}
-
-/// Message payload definition for restart operation request.
-#[derive(Debug, Deserialize, Serialize, Eq, PartialEq)]
-#[serde(deny_unknown_fields)]
-#[serde(rename_all = "camelCase")]
-pub struct RestartOperationRequest {
-    pub id: String,
+/// Command to restart a device
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct RestartCommand {
+    pub target: EntityTopicId,
+    pub cmd_id: String,
+    pub payload: RestartCommandPayload,
 }
 
-impl<'a> Jsonify<'a> for RestartOperationRequest {}
+impl RestartCommand {
+    /// Create a [RestartCommand] to restart a target device.
+    ///
+    /// - use a fresh cmd id
+    /// - set the status to [CommandStatus::Init]
+    pub fn new(target: EntityTopicId) -> Self {
+        let cmd_id = nanoid!();
```

Comment:
~Wouldn't it be better to pass that `cmd_id` as an argument, so that the caller has better control on whether to create unique IDs or pass something know? Especially for unit tests.~

Saw the `with_id` function below that solves this problem. This is better.